File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>m;
        int found;
        for(int i=0;i<nums.size();i++){
            found = target-nums[i];
            if(m.count(found)){
                return {m[found],i};
            }
            else{
                m[nums[i]] = i;
            }
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let index = 0
    let mp = new Map();
    for(const val of nums){
        const find = target - val;
        if(mp.has(find)){
            return [mp.get(find),index];
        }
        else mp.set(val,index);
        index++;
    }
    return [-1,-1];
};